### ISSUE TITLE
Investigate `dc` vs `rc` modes on InfiniBand

### ISSUE THREAD

**Issue by pentschev** [rapidsai/dask-cuda#215]
According to [Connect-IB™: Architecture for Scalable High Performance Computing](http://www.mellanox.com/related-docs/applications/SB_Connect-IB.pdf) solution brief from Mellanox, using the `rc` mode (currently hardcoded in dask-cuda) may become an intractable memory problem as we scale systems. We need to investigate how `rc` and `dc` modes affect performance and memory consumption in dask-cuda.

cc @jakirkham @quasiben

**Comment by jakirkham** [comment]
Have we had a chance to explore this?

**Comment by pentschev** [comment]
No, I haven't explored this, perhaps it's something we could do this year.
